lib/authlogic/session/password.rb

```diff
@@ -154,12 +154,22 @@ def credentials
         # hash form.
         def credentials=(value)
           super
-          values = parse_param_val(value) # add strong parameters check
-
-          if values.first.is_a?(Hash)
-            values.first.with_indifferent_access.slice(login_field, password_field).each do |field, value|
-              next if value.blank?
-              send("#{field}=", value)
+          values = value.is_a?(Array) ? value.first : value
+
+          if values.respond_to? :keys
+            values = values.with_indifferent_access if values.is_a?(Hash)
+            values = values.slice(login_field, password_field)
+            missing_values = [login_field.to_s, password_field.to_s] - values.keys.map(&:to_s)
+
+            if missing_values.empty?
+              values.each do |field, value|
```

Comment:
Block variable `value` shadows argument also named `value`. Please rename block variable to something like `val` to avoid confusion.